controllers/tempest_controller.go

```diff
@@ -515,6 +512,34 @@ func mergeWithWorkflow[T any](value T, workflowValue *T) T {
 	return *workflowValue
 }
 
+func getTempestRun(instance *testv1beta1.Tempest, workflowStepNum int) testv1beta1.TempestRunSpec {
```

Comment:
All the newly introduced functions should be moved to `common.go`.

[1] https://github.com/openstack-k8s-operators/test-operator/blob/main/controllers/common.go